catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardTypeFactory.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.data.impl;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.MetacardType;
+
+public class MetacardTypeFactory {
+
+    public static MetacardType createMetacardType(String name, List<MetacardType> typeList) {
```

Comment:
I agree that the MetacardType isn't necessarily a composition of other MetacardTypes.  I'm in favor of using a new interface like `MetacardComponent` to define these categories in the new Taxonomy.  A list of  `MetacardComponents` seems more appropriate here.